File: Car Cloth/car-cloth-web/src/constants/TransactionStatus.js

export const TRANSACTION_STATUS = {
    COMPARE: {
        label: '견적 비교',
        value: 'COMPARE'
    },
    RESERVATION: {
        label: '예약',
        value: 'RESERVATION',
    },
    CONSTRUCTING: {
        label: '시공 중',
        value: 'CONSTRUCTING',
    },
    COMPLETE: {
        label: '시공 완료',
        value: 'COMPLETE',
    }
}

export const TRANSACTION__OBJECT_STATUS = {
    'COMPARE': '견적 비교',
    'RESERVATION': '예약',
    'CONSTRUCTING': '시공 중',
    'COMPLETE': '시공 완료'
}

// export const STEPS_TRANSACTION_STATUS = ['COMPARE', 'RESERVATION', 'CONSTRUCTING', 'COMPLETE']
export const STEPS_TRANSACTION_STATUS = Object.keys(TRANSACTION__OBJECT_STATUS)

File: Car Cloth/car-cloth-backend/src/main/java/com/mintpot/carcloth/constant/TransactionStatus.java
package com.mintpot.carcloth.constant;

public enum TransactionStatus {
    COMPARE(TransactionStatus.Constant.COMPARE),
    RESERVATION(TransactionStatus.Constant.RESERVATION),
    CONSTRUCTING(TransactionStatus.Constant.CONSTRUCTING),
    COMPLETE(TransactionStatus.Constant.COMPLETE);

    private final int code;

    TransactionStatus(final int code) {
        this.code = code;
    }

    public final int getCode() {
        return this.code;
    }

    public static TransactionStatus valueOf(int code) {
        for (TransactionStatus type : TransactionStatus.values()) {
            if (type.getCode() == code) {
                return type;
            }
        }

        throw new IllegalArgumentException(TransactionStatus.class.getName() + " does not have value with code: " + code);
    }

    public static class Constant {
        public static final int COMPARE = 0;
        public static final int RESERVATION = 2;
        public static final int CONSTRUCTING = 5;
        public static final int COMPLETE = 9;
    }
}
